src/LegendList.tsx

```diff
@@ -792,25 +794,18 @@ const LegendListInner = typedForwardRef(function LegendListInner<T>(
         }
     };
 
-    const isFirst = !refState.current.renderItem;
-    // Run first time and whenever data changes
-    if (isFirst || dataProp !== refState.current.data || numColumnsProp !== peek$<number>(ctx, "numColumns")) {
-        if (!keyExtractorProp && !isFirst && dataProp !== refState.current.data) {
-            // If we have no keyExtractor then we have no guarantees about previous item sizes so we have to reset.
-            // Note: don't want to clear sizes because if sizes don't change they won't update from onLayout
-            // so it's safer to fallback to previous sizes than to the estimate.
-            refState.current.positions.clear();
-        }
-
-        refState.current.data = dataProp;
-
+    const calcTotalSizes = (forgetPositions = false) => {
```

Comment:
@jmeistrich not sure if that's best name for the function, maybe you have suggestion